### ISSUE TITLE
Fix S1854 FP: Support newlines in string interpolation

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6302]
### Description

FP for S1854 (DeadStores).

Using C#11 newlines in string interpolation the rule is raising an issue.

### Repro steps

```csharp
string empty = "";
    string x = $"{empty +
        empty}"; // Noncompliant@-1 FP (string is still empty, should be compliant)
    x = "Test";
    Foo(x);
```

### Expected behavior

The rule should not raise an issue.

### Actual behavior

The rule does raise an issue.

**Comment by csaba-sagi-sonarsource** [comment]
The rule is still going to raise for these values as we intentionally don't want to evaluate the string interpolations to look for empty strings.
